packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -775,6 +742,43 @@ export function clearContainer(container: Container): void {
   }
 }
 
+function clearContainerChildren(container: Node) {
+  let node;
+  let nextNode: ?Node = container.firstChild;
+  while (nextNode) {
+    node = nextNode;
+    nextNode = nextNode.nextSibling;
+    switch (node.nodeName) {
+      case 'HTML':
+      case 'HEAD':
+      case 'BODY': {
+        const element: Element = (node: any);
+        clearContainerChildren(element);
+        // If these singleton instances had previously been rendered with React they
+        // may still hold on to references to the previous fiber tree. We detatch them
+        // prospectively to reset them to a baseline starting state since we cannot create
+        // new instances.
+        detachDeletedInstance(element);
+        continue;
+      }
+      case 'STYLE': {
+        continue;
+      }
+      case 'LINK': {
+        if (((node: any): HTMLLinkElement).rel.toLowerCase() === 'stylesheet') {
+          continue;
+        }
+      }
+    }
+    if (node.nodeType === DOCUMENT_TYPE_NODE || isMarkedResource(node)) {
```

Comment:
Do we really need to keep these? Aren't we just going to recreate them anyway? It's kind of an error state.

Reply:
It was useful for stylesheets due to FUOC but for the rest.

Reply:
Well the doctype doesn’t get recreated but I think you meant the resource check.

yeah I think I may have added this unneccessarily. It’s important for clearSingleton because acquireResource can happen before clearSingleton. But for clearContainer that’s not really an issue. I’ll remove